### ISSUE TITLE
ML Model for training on image digital signatures and image verifications

### ISSUE THREAD

**Issue by yuvraajnarula** [yuvraajnarula/xplore-ieee#1]
Objective

Implement an ML + cryptographic pipeline to generate digital signatures for images and perform verification (identity/provenance + tamper detection) as part of the Zillionid MVP.

Procedure
1. Image Canonicalization (Free & Deterministic)

-  Convert to sRGB (via OpenCV/Pillow – both free).
-  Strip EXIF/metadata (Pillow).
-  Resize to fixed size (e.g., 1024 px max) with bilinear interpolation.
-  Save as PNG (lossless, free).

2. Digital Signature (Free Cryptography)

-  Compute hashes:SHA-256 (from Python hashlib) → cryptographic hash.
- Perceptual Hash (pHash) → use [ImageHash](https://pypi.org/project/ImageHash/)
-  (MIT License).
-  Apply Post-Quantum Signature:
 - Use [liboqs](https://github.com/open-quantum-safe/liboqs)
-  (Apache 2.0) or [pqcrypto](https://pypi.org/project/pqcrypto/)
-  for Dilithium signatures.
-  Store {sha256_digest, phash, pqc_signature, pubkey_id} in wallet/ledger.

3. Feature Embedding Model (Free Pre-trained Models)

-  Use torchvision.models (all under BSD-style license):
- ResNet50, ResNet18, MobileNetV3 (all free).
-  Fine-tune for embeddings with contrastive loss or triplet loss (no proprietary loss functions required).
-  Store 128–256 dim embeddings in ledger (FAISS or Annoy, both free for search).

4. Manipulation Detector (Free Datasets + Models)

- Model: Xception (Keras/PyTorch free impl) or ResNet18.
 - Datasets (all free for research/commercial):
- CASIA 2.0 Tampered Image Dataset
- Columbia Image Splicing Dataset

- [FaceForensics++](https://github.com/ondyari/FaceForensics)
 (free academic use, but check license if commercial).
- Optionally: generate synthetic manipulations yourself (crop, blur, JPEG re-encode — all free).
- Output: probability of manipulation.

5. Verification Workflow

- Input image → Canonicalize.
 - Run cryptographic verification:
- PQC_verify(pubkey, signature, sha256_digest)

If valid → mark high trust.

 If invalid or absent:

Compute embedding → cosine similarity with stored reference.
- Run manipulation detector.
- Combine into trust score:
- trust = `w1*crypto_status + w2*similarity + w3*(1 - manip_prob)`
- Normalize with entropy-based scoring.

6. Evaluation (Free Metrics)

- Metrics: ROC-AUC, Equal Error Rate (EER), False Accept/Reject Rate (all standard).
- Validate on free test datasets and synthetic corruptions (JPEG, crops, noise).

7. Integration (All Free/Open)

-  Wrap inference in FastAPI (MIT License). 
-  Return JSON response

```json
{
  "crypto_status": true,
  "similarity_score": 0.92,
  "manipulation_score": 0.08,
  "trust_score": 0.87
}
